### PR TITLE
Fix export endpoint #28

### LINKED ISSUE #28: Export endpoint fails currently
I think with the new updated Zotero version, which we use now here in translation-server, the `export` endpoint is not working anymore. The following error message will occur:

```
zotero(3)(+0000002): Zotero.Utilities.itemToExportFormat is not a function at chrome://translation-server/content/translate_item.js:52
```

This function [`itemToExportFormat`](https://github.com/zotero/zotero/blob/master/chrome/content/zotero/xpcom/utilities_internal.js#L768) changed its location to `Zotero.Utilities.Internal` with https://github.com/zotero/zotero/commit/47bf9c38e96e41d066a1c267365a6010be42b038. I tried to replace that and load that file in the same way as `utilities.js` in `build.sh` and `init.js`, but this didn't work out for me :-( Actually, the building process looks ok, but then running the translation-server will just silently quit.

### PR DESCRIPTION
**Pull request by zuphilip**
This fixes #28 i.e. the export endpoint which is currently broken.

I tested it with (in all formats):

```
curl -X POST --header 'Content-Type: application/json' --header 'Accept: text/plain' -d '  [{"itemKey":"9XZWACW9","itemVersion":0,"itemType":"bookSection","creators":[{"firstName":"Rita","lastName":"Banzi","creatorType":"author"},{"firstName":"Cristina","lastName":"Cusi","creatorType":"author"},{"firstName":"Concetta","lastName":"Randazzo","creatorType":"author"},{"firstName":"Roberto","lastName":"Sterzi","creatorType":"author"},{"firstName":"Dario","lastName":"Tedesco","creatorType":"author"},{"firstName":"Lorenzo","lastName":"Moja","creatorType":"author"}],"tags":[],"title":"Selective serotonin reuptake inhibitors (SSRIs) and serotonin‐norepinephrine reuptake inhibitors (SNRIs) for the prevention of tension‐type headache in adults","bookTitle":"The Cochrane Library","publisher":"John Wiley & Sons, Ltd","language":"en","date":"2015/05/01","url":"http://onlinelibrary.wiley.com/doi/10.1002/14651858.CD011681/abstract","accessDate":"CURRENT_TIMESTAMP","libraryCatalog":"Embedded Metadata"}]
' 'http://localhost:1969/export?format=ris'
```

All formats except `tei` and `mods` are working (there is another problem with the XML export translators, which we can try to fix later).

### CODE REVIEW

**Comment by kba** [comment]
`itemToExportFormat` has support for backwards-compatibility transformations in  [addCompatibilityMappings](https://github.com/zotero/zotero/blob/7f339af1042b36362912832e6c6e506e4dec01c8/chrome/content/zotero/xpcom/utilities_internal.js#L769). I don't know how many translators this concerns but with this fix, this step is skipped.

**Comment by zuphilip** [comment]
It might be that we are losing something here... But I thought it might be easier to discuss if we have a PR (and I will thereby not forget it in the next days). Maybe, there are better solutions to do.

The compatibility mapping was introduced in PR https://github.com/zotero/zotero/pull/659 where it is mentioned that:

> The idea is that new export translators (and old translators as they are slowly transitioned) should only rely on the serialize format. Once that is done, we can remove the compatibility mappings.

I can't remember any fixing since May 2015 in the translators. But I am also not sure what these issues would look like in the translator codes.

As far as I understand the `itemToExportFormat` will only be called in the export translators by looping with `nextItem`. Maybe, the question is also what do we expect in translation-server as input for the `export` endpoint?

File: src/translate_item.js [hunk 1]
```diff
@@ -49,8 +49,14 @@ Zotero.Translate.ItemGetter.prototype = {
 	 */
 	"nextItem":function() {
 		if(!this._itemsLeft.length) return false;
-		var item = Zotero.Utilities.itemToExportFormat(this._itemsLeft.shift());
-		
+		var item = this._itemsLeft.shift();
+		if (!item.attachements) {
```
Comment (dstillman) [inline src/translate_item.js]: Typo

File: build.sh [hunk 2]
```diff
@@ -89,4 +89,4 @@ echo "resource zotero translation-server/resource/" >> "$BUILDDIR/chrome.manifes
 
 find "$BUILDDIR" -depth -type d -name .svn -exec rm -rf {} \;
 find "$BUILDDIR" -name .DS_Store -exec rm -rf \;
-find "$BUILDDIR" -name '._*' -exec rm -rf \;
\ No newline at end of file
+find "$BUILDDIR" -name '._*' -exec rm -rf \;
```
Comment (dstillman) [inline build.sh]: `git commit -p` the unrelated lines

**Comment by dstillman** [comment]
> Maybe, the question is also what do we expect in translation-server as input for the export endpoint?

It [gets](https://github.com/zotero/dataserver/blob/6e3b732e64590fa98e5dd69e33d6ec8b7bed73f8/model/Translate.inc.php#L49) current API JSON (plus a `uri` field) as input. The compatibility mappings are for the client, and that affects a few fields (e.g., `version` in `computerProgram`), but translators [should just be updated](https://groups.google.com/d/msg/zotero-dev/FWCP_R5S_hg/ditgqvOpDwAJ) to work with the API/translation-server.

So I think this is the right fix, though I haven't tested it yet.

File: src/translate_item.js [hunk 3]
```diff
@@ -49,8 +49,14 @@ Zotero.Translate.ItemGetter.prototype = {
...
+			item.attachments = [];
+		}
+		if (!item.notes) {
+			item.notes = [];
+		}
+
 		// convert single field creators to format expected by export
 		if(item.creators) {
 			for(var i=0; i<item.creators.length; i++) {
```
Comment (zuphilip) [inline src/translate_item.js:62]: Should I delete this conversion of some legacy format as well? Is this still important for current API JSON format? I think this is the same as [here](https://github.com/zotero/zotero/blob/4.0/chrome/content/zotero/xpcom/utilities_internal.js#L466-480) where it is written:

> // Old format used to supply creatorID (the database ID), but no
>                     // translator ever used it

File: src/translate_item.js (hunk 1 above)
Comment (zuphilip) [inline src/translate_item.js]: Fixed (this typo is haunting me:)

File: build.sh (hunk 2 above)
Comment (zuphilip) [inline build.sh]: I deleted the new lines (for this PR) from `vim` on Linux with switching to Notepad++ on Windows.

File: src/translate_item.js (hunk 3 above)
Comment (dstillman) [inline src/translate_item.js:62]: No, it looks like the export translators (at least RIS and BibTex) haven't actually been updated and still depend on the old internal creator format (`firstName`, `lastName`, `fieldMode`=0/1) as opposed to the API format (`firstName`/`lastName` or `name`). So this probably needs to stay in for now for single-field creators to work properly.

**Comment by zuphilip** [comment]
You can test this branch yourself with just these two lines (just make sure you are in a location where larger files are no problem for your internet connection):

```
docker run -d -p 1969:1969 zuphilip/translation-server:export

curl -X POST --header 'Content-Type: application/json' --header 'Accept: text/plain' -d '  [{"itemKey":"9XZWACW9","itemVersion":0,"itemType":"bookSection","creators":[{"firstName":"Rita","lastName":"Banzi","creatorType":"author"},{"firstName":"Cristina","lastName":"Cusi","creatorType":"author"},{"firstName":"Concetta","lastName":"Randazzo","creatorType":"author"},{"firstName":"Roberto","lastName":"Sterzi","creatorType":"author"},{"firstName":"Dario","lastName":"Tedesco","creatorType":"author"},{"firstName":"Lorenzo","lastName":"Moja","creatorType":"author"}],"tags":[],"title":"Selective serotonin reuptake inhibitors (SSRIs) and serotonin‐norepinephrine reuptake inhibitors (SNRIs) for the prevention of tension‐type headache in adults","bookTitle":"The Cochrane Library","publisher":"John Wiley & Sons, Ltd","language":"en","date":"2015/05/01","url":"http://onlinelibrary.wiley.com/doi/10.1002/14651858.CD011681/abstract","accessDate":"CURRENT_TIMESTAMP","libraryCatalog":"Embedded Metadata"}]
' 'http://localhost:1969/export?format=ris'
```

File: src/translate_item.js (hunk 3 above)
Comment (zuphilip) [inline src/translate_item.js:62]: Okay, I see.

**Pull request merged**